### PR TITLE
Update dnspolicy crd validations

### LINKED ISSUE #554: DNSPolicy: defaultGeo should be required in the DNSPolicy spec
**What**

The `defaultGeo` value in `spec.loadBalancing.geo` should be required, and it should not be possible to create a geo section without providing it with a non zero length string.

Invalid:
```
spec:
  loadBalancing:
    geo: {}
  routingStrategy: loadbalanced
  targetRef:
    group: gateway.networking.k8s.io
    kind: Gateway
    name: distributed-us
```

It should also handle better the case where the geo for whatever reason is calculated to be an empty string. Currently it will create invalid dnsNames like `.klb.distributed.managed.hcpapps.net` where it should be `<geocode>.klb.distributed.managed.hcpapps.net`.

### PR DESCRIPTION
**Pull request by mikenairn**
closes #554 

[fix: DNSPolicy CRD required validations](https://github.com/Kuadrant/kuadrant-operator/commit/85e99a8b8e3b86899b4c5f6a5aafb5d81f85fb40)

The validations to ensure required fields were marked as required were incorrect.
Any fields that are not explicitly marked with `+optional` and do not have `omitempty` are considered required.

[Update loadbalanced routingStrategy CRD validation](https://github.com/Kuadrant/kuadrant-operator/commit/fdf78056bff073cfd2cf94e85d91e6a2c3537847)

Update the CRD validations to make the "loadBalancing" field required if the routingStrategy is "loadbalanced" and make the "defaultWeight" and "defaultGeo" fields mandatory within the "loadBalancing" section.

Minimum valid specs are now:

Simple:
```
spec:
  targetRef:
    name: prod-web
    group: gateway.networking.k8s.io
    kind: Gateway
  routingStrategy: simple
```

LoadBalanced:
```
spec:
  targetRef:
    name: prod-web
    group: gateway.networking.k8s.io
    kind: Gateway
  routingStrategy: loadbalanced
  loadBalancing:
    geo:
      defaultGeo: foo
    weighted:
      defaultWeight: 100
```

### CODE REVIEW

File: controllers/dnspolicy_controller_multi_cluster_test.go [hunk 1]
```diff
@@ -183,148 +183,6 @@ var _ = Describe("DNSPolicy Multi Cluster", func() {
 
 	Context("loadbalanced routing strategy", func() {
 
-		Context("weighted", func() {
```
Comment (mikenairn) [inline controllers/dnspolicy_controller_multi_cluster_test.go:186]: Irrelevant, you can't have weighted without geo.

Comment (makslion) [reply controllers/dnspolicy_controller_multi_cluster_test.go:186]: i'm removing that one in #549

File: controllers/dns_helper.go [hunk 2]
```diff
@@ -208,7 +208,6 @@ func (dh *dnsHelper) getLoadBalancedEndpoints(mcgTarget *multicluster.GatewayTar
 	}
 
 	var endpoint *externaldns.Endpoint
-	var defaultEndpoint *externaldns.Endpoint
```
Comment (mikenairn) [inline controllers/dns_helper.go:211]: @makslion fyi, i removed this as it's not needed any longer. Also added some more relevant comments below.

**Review by maleck13 — APPROVED** [review]

**Comment by mikenairn** [comment]
Test are still broken for other reasons, verified the dns tests locally:

```
./bin/ginkgo --focus "DNSPolicy" --focus "TLSPolicy" --skip "Target status reconciler" -tags integration ./controllers/...
Running Suite: Controller Suite - /home/mnairn/go/src/github.com/kuadrant/kuadrant-operator/controllers
=======================================================================================================
Random Seed: 1714118795

Will run 31 of 147 specs
•••••••••••••••SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS••••••••••••••••SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 31 of 147 Specs in 255.786 seconds
SUCCESS! -- 31 Passed | 0 Failed | 0 Pending | 116 Skipped
PASS

Ginkgo ran 1 suite in 4m23.97465344s
Test Suite Passed
```

**Pull request merged**